### PR TITLE
Minimise DB calls to throw duplicate app name error message

### PR DESCRIPTION
**Pull request by abhvsn**
## Description

Currently to report error message for updating duplicate application names within the organization takes 3 DB calls which is minimized to a single one.  

## Type of change

- Enhancement feature (non-breaking change which adds functionality)

## How Has This Been Tested?

> Tested manually by trying to duplicate the application name within the same organization 
> Junit TC

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

## Test coverage results :test_tube:
<details><summary>:white_circle: Total coverage has not changed</summary>
No changes to code coverage between the base branch and the head branch</details>

### CODE REVIEW

File: app/client/src/components/designSystems/blueprint/TextComponent.tsx [hunk 1]
```diff
@@ -64,6 +64,7 @@ export interface TextComponentProps extends ComponentProps {
   backgroundColor?: string;
   textColor?: string;
   fontStyle?: string;
+  renderAsHTML?: boolean;
```
Comment (mohanarpit) [inline app/client/src/components/designSystems/blueprint/TextComponent.tsx:67]: Is this change required as part of this PR? Seems to be code from another feature.

File: app/server/appsmith-server/src/main/java/com/appsmith/server/services/ApplicationServiceImpl.java [hunk 2]
```diff
@@ -128,17 +127,19 @@ public ApplicationServiceImpl(Scheduler scheduler,
...
-        return isNameAllowed(id, application)
-            .flatMap(nameAllowed -> {
-                // If the name is allowed, return update resource for further processing
-                if (Boolean.TRUE.equals(nameAllowed)) {
-                    return repository.updateById(id, application, AclPermission.MANAGE_APPLICATIONS)
-                        .flatMap(analyticsService::sendUpdateEvent);
+        return repository.updateById(id, application, AclPermission.MANAGE_APPLICATIONS)
+            .onErrorResume(error -> {
+                if (error instanceof DuplicateKeyException) {
+                    if (error.getCause().getMessage().contains("name:")) {
```
Comment (mohanarpit) [inline app/server/appsmith-server/src/main/java/com/appsmith/server/services/ApplicationServiceImpl.java:133]: Can you please add a comment with an example error message returned by MongoDB and what will this error get formatted into? I'm a little unsure of this based on purely reading the code.

File: app/server/appsmith-server/src/test/java/com/appsmith/server/services/ApplicationServiceTest.java [hunk 3]
```diff
@@ -328,8 +328,7 @@ public void invalidUpdateApplication() {
             .flatMap(t -> applicationService.update(t.getId(), t));
 
         StepVerifier.create(updateInvalidApplication)
-            .expectErrorMatches(throwable -> throwable instanceof AppsmithException &&
-                throwable.getMessage().equals(AppsmithError.DUPLICATE_KEY_USER_ERROR.getMessage(testApp1.getName(), FieldName.NAME)))
+            .expectErrorMatches(throwable -> throwable instanceof AppsmithException)
```
Comment (mohanarpit) [inline app/server/appsmith-server/src/test/java/com/appsmith/server/services/ApplicationServiceTest.java:331]: Why not assert the exception message thrown as well?

Comment (abhvsn) [reply app/server/appsmith-server/src/test/java/com/appsmith/server/services/ApplicationServiceTest.java:331]: When I'm trying to run the TC I'm not getting the keys in the error string on which we are deciding if the error is related to a duplicate name for the application.

Error message for update request through API : _E11000 duplicate key error collection: appsmith.application index: organization_application_deleted_compound_index dup key: { organizationId: "***a320f25346f6d990e***", name: "Untitled application 1", deletedAt: null }_

Error message from TC : _E11000 duplicate key error collection: appsmith.application index: organization_application_deleted_compound_index dup key: { : "***3fa08ce1bb74c04cdf***", : "validApplication1", : null }_

Because of this, we are getting a different error message here.

File: app/server/appsmith-server/src/main/java/com/appsmith/server/services/ApplicationServiceImpl.java (hunk 2 above)
Comment (mohanarpit) [reply app/server/appsmith-server/src/main/java/com/appsmith/server/services/ApplicationServiceImpl.java:136]: Through the API, what is the formatted error message returned to the client look like?

**Review by mohanarpit — APPROVED** [review]
